model/src/main/kotlin/utils/FindingsMatcher.kt

```diff
@@ -274,5 +280,67 @@ fun associateLicensesWithExceptions(
         exception.copy(license = "${SpdxConstants.NOASSERTION} ${SpdxExpression.WITH} ${exception.license}".toSpdx())
     }
 
-    return fixedLicenses
+    return fixedLicenses.mapTo(mutableSetOf()) { it.copy(license = associateLicensesWithExceptions(it.license)) }
+}
+
+/**
```

Comment:
This function to me seems quite complex.
I fear it can happen relatively easily to introduce some bug here.
As the function is underneath the call tree of `toSummary` of `ScanResultModelMapper`,
the returned summary may get stored in a scan  result storage. 

So if it turns out there is a bug in, then the stored scan results become kind of invalid.
I believe this is a bit risky, what do you think?

Reply:
> I fear it can happen relatively easily to introduce some bug here.

I agree that the code is rather complex, and that there could be lingering bugs, like with any code that we write.

> the returned summary may get stored in a scan result storage.

Yeah, we bump into this topic on a regular basis. Storing or not storing the results of post-processing scan results in the scan storage both have its pros and cons. However, as the result of associating exceptions across *different* findings already is stored as well, I don't think it makes a difference here.

> I believe this is a bit risky, what do you think?

Generally, I don't think it's too risky to have such code. To me, the scan storage is mostly a cache for speeding up scans, not a database of definitely correct license information. And as such it's fine to update the storage with corrections if needed. The bigger problem IMO would be with wrong stored scan results going unnoticed. But this is more unlikely to happen, I believe.

Reply:
I just wanted to make sure it is considered by the devs approving this.